### ISSUE TITLE
bug alert!

### ISSUE THREAD

**Issue by asobhy-qnx** [eclipse-cyclonedds/cyclonedds#807]
https://github.com/eclipse-cyclonedds/cyclonedds/blob/6e16753f971049061a27fd70c70e2d780ff321ef/src/core/ddsi/src/q_ddsi_discovery.c#L73

should be

```
(ddsi_is_mcaddr (gv, &loc->c) && !(allow_multicast & DDSI_AMC_ASM)
```

**Comment by k0ekk0ek** [comment]
Thanks for reporting @asobhy-qnx! @eboasson, you want to include this in ##805?

**Comment by eboasson** [comment]
Fixed (#814)
